packages/wallet/src/KeyManagement/LedgerKeyAgent.ts

```diff
@@ -164,11 +170,68 @@ export class LedgerKeyAgent extends KeyAgentBase {
     });
   }
 
+  async signTransaction(initializedTxData: InitializeTxResult): Promise<Cardano.Signatures> {
+    try {
+      const deviceConnection = await LedgerKeyAgent.checkDeviceConnection(this.serializableData.communicationType, this.deviceConnection);
+      const dataForSigning = this.#prepareDataForSigning(initializedTxData);
+      const response = await deviceConnection.signTransaction(dataForSigning);
+      // TODO - derive witnes(es) path(s) and return Map<Ed25519PublicKey, Ed25519Signature>
+      return response.witnesses;
+    } catch (error: any) {
+      // eslint-disable-next-line unicorn/numeric-separators-style
+      if (error.code === 28169) {
+        throw new AuthenticationError('Transaction signing aborted', error);
+      }
+      throw new TransportError('Transport failed', error);
+    }
+    return response;
+  }
+
   async signBlob(): Promise<SignBlobResult> {
     throw new NotImplementedError('signBlob');
   }
 
   async exportRootPrivateKey(): Promise<Cardano.Bip32PrivateKey> {
     throw new NotImplementedError('Operation not supported!');
   }
+
+  async #prepareDataForSigning({ body, hash, inputSelection }: InitializeTxResult) {
```

Comment:
Should probably explicitly specify the return type from ledger library. Same when converting each transaction property.

Reply:
@mkazlauskas aha, I know. I have that in my TODO list.